src/app/components/annotations/components/grid-tile-content/grid-tile-content.component.ts

```diff
@@ -0,0 +1,114 @@
+import type { SpectrogramComponent } from "@ecoacoustics/web-components/@types/components/spectrogram/spectrogram";
+import type { MediaControlsComponent } from "@ecoacoustics/web-components/@types/components/media-controls/media-controls";
+import { Annotation } from "@models/data/Annotation";
+
+export const gridTileContextSelector = "baw-grid-tile-content" as const;
+
+@Component({
+  standalone: true,
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+  encapsulation: ViewEncapsulation.ShadowDom,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+
+  selector: "baw-ng-grid-tile-content",
+  templateUrl: "grid-tile-content.component.html",
+  styleUrl: "grid-tile-content.component.scss",
+})
+export class GridTileContentComponent implements AfterViewInit {
+  @ViewChild("wrapper")
+  private wrapper: ElementRef<HTMLDivElement>;
+
+  @ViewChild("context-spectrogram")
+  private contextSpectrogram: ElementRef<SpectrogramComponent>;
+
+  @ViewChild("context-media-controls")
+  private contextMediaControls: ElementRef<MediaControlsComponent>;
+
+  protected model = signal<Annotation>(undefined);
+  protected contextExpanded = false;
+
+  public get listenLink(): string {
+    if (!this.model()) {
+      return "";
+    }
+
+    return this.model().viewUrl;
+  }
+
+  public get contextSpectrogramId(): string {
+    return `spectrogram-${this.model().id}-context`;
+  }
+
+  public get contextSource(): string {
+    if (!this.model()) {
+      return "";
+    }
+
+    const contextSize = 30;
+
+    const startTime = this.model().startTimeSeconds;
+    const endTime = this.model().endTimeSeconds;
+
+    // TODO: the end time should check that it is less than the duration of the audio file
+    const newStart = Math.max(0, startTime - contextSize);
+    const newEnd = endTime + contextSize;
+
+    const existingUrl = new URL(this.model().audioLink);
+    const searchParams = new URLSearchParams(existingUrl.search);
+
+    searchParams.set("start_offset", newStart.toString());
+    searchParams.set("end_offset", newEnd.toString());
+    searchParams.delete("audio_event_id");
+
+    const newUrl = new URL(existingUrl.origin + existingUrl.pathname);
+    newUrl.search = searchParams.toString();
+
+    return newUrl.toString();
+  }
+
+  public ngAfterViewInit(): void {
+    this.requestContext();
+  }
+
+  public handleContextChange(subjectWrapper: SubjectWrapper): void {
+    this.model.set(new Annotation(subjectWrapper.subject));
+  }
+
+  protected toggleContext(): void {
+    this.contextExpanded = !this.contextExpanded;
+  }
+
+  private async requestContext(): Promise<void> {
+    // TODO: remove this once @ecoacoustics/web-components#219 is resolved
+    const webComponents = await import("@ecoacoustics/web-components")
+    const token = webComponents.gridTileContext;
+
+    this.wrapper.nativeElement.dispatchEvent(
+      new ContextRequestEvent(
+        token,
+        this.handleContextChange.bind(this),
+        true
+      )
+    );
```

Comment:
we should bind this up into a decorator on a field just like `@ViewElementRef` (TODO)

Reply:
Something like this seems to do the trick

![image](https://github.com/user-attachments/assets/8df8914e-40fc-44bd-8684-ce5b70223945)